File: leetcode/128. Longest Consecutive Sequence/ans.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest_streak = 0
        numberSet = set(nums)
        
        for num in numberSet:
            if num - 1 not in numberSet:
                currentNum = num
                currentStreak = 1
                
                while currentNum + 1 in numberSet:
                    currentNum += 1
                    currentStreak += 1
                
                longest_streak = max(longest_streak, currentStreak)
                
        return longest_streak

File: leetcode/128. Longest Consecutive Sequence/ans.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        if(nums.size() == 0)
            return 0;
        
        int longestStreak = 0;
        
        
        map<int,int>mp;
        
        for(int i = 0; i < nums.size(); i++)
        {
            mp[nums[i]];
        }
        

        
        int streak = 1;
        auto it2 = mp.begin();
        for(auto it = mp.begin(); it!=mp.end();it++)
        {
            if(it->first-1 == it2->first)
            {
                 streak++;                
            }
            else
            {
                if(longestStreak < streak)
                    longestStreak = streak; 
                streak = 1;
                               
            }

            
            it2 = it;
        }
        if(longestStreak < streak)
            longestStreak = streak; 
        return longestStreak;
    }
};
